lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -120,6 +122,35 @@ class QcFlagService {
         return dplDetector;
     }
 
+    /**
+     * Update given effective periods of some QC flags
+     * @param {Period} intersectingPeriod newer QC flag period
```

Comment:
No notion of QC flag here, and no notion of newer. It's simply the period that need to be removed from the list of effective perids.

The functions need to be named and documented based on what they do, not on where you expect them to be used

Reply:
applied